sklearn/covariance/_shrunk_covariance.py

```diff
@@ -325,31 +309,14 @@ def ledoit_wolf(X, *, assume_centered=False, block_size=1000):
 
     where mu = trace(cov) / n_features
     """
-    X = check_array(X)
-    # for only one feature, the result is the same whatever the shrinkage
-    if len(X.shape) == 2 and X.shape[1] == 1:
-        if not assume_centered:
-            X = X - X.mean()
-        return np.atleast_2d((X**2).mean()), 0.0
-    if X.ndim == 1:
-        X = np.reshape(X, (1, -1))
-        warnings.warn(
-            "Only one sample available. You may want to reshape your data array"
-        )
-        n_features = X.size
-    else:
-        _, n_features = X.shape
+    X = as_float_array(X)
 
-    # get Ledoit-Wolf shrinkage
-    shrinkage = ledoit_wolf_shrinkage(
-        X, assume_centered=assume_centered, block_size=block_size
-    )
-    emp_cov = empirical_covariance(X, assume_centered=assume_centered)
-    mu = np.sum(np.trace(emp_cov)) / n_features
-    shrunk_cov = (1.0 - shrinkage) * emp_cov
-    shrunk_cov.flat[:: n_features + 1] += shrinkage * mu
+    estimator = LedoitWolf(
+        assume_centered=assume_centered,
+        block_size=block_size,
+    ).fit(X)
 
-    return shrunk_cov, shrinkage
```

Comment:
I would isolate this part of the code in a `_ledoit_wolf` private function for the moment.

The only thing that we can now assume is that `X` has been validated and it is 2d. So we can use safely `n_features = X.shape[1]`.